### ISSUE TITLE
ADR-0001: Per-target readouts — Salience, Anomaly, Pattern

### ISSUE THREAD

**Issue by salvatorebottiglieri** [salvatorebottiglieri/botti-agent#10]
## Parent

ADR-0001: Reservoir Computing for the Learning Module

## What to build

Three linear readouts on top of the ESN, each targeting a different downstream signal.

**Deliverables:**
- **Salience readout** (`SalienceReadout`)
  - Input: reservoir state vector
  - Target: event importance score (0–1)
  - `is_salient(event)` returns bool based on readout threshold
  - Used by Wave 7.1
- **Anomaly readout** (`AnomalyReadout`)
  - Input: reservoir state vector
  - Target: deviation from learned normal pattern
  - `is_anomalous(state)` returns bool based on reconstruction error
  - Used by Wave 7.2
- **Pattern readout** (`PatternReadout`)
  - Input: reservoir state vector
  - Target: pattern class probabilities (multi-class)
  - `get_pattern_probabilities(state)` returns dict of class to probability
  - Used by Wave 7.3
- Shared `EchoStateNetwork` instance (same reservoir state for all three)
- Each readout trained independently via ridge regression
- Tests: `tests/learning/test_readouts.py`
  - Each readout trains and predicts on synthetic labeled data
  - Cross-readout independence confirmed

**Boundaries:**
- No event-bus integration yet
- Readouts only compute from reservoir state, no external data

## Acceptance criteria

- [ ] SalienceReadout identifies salient events correctly on synthetic data
- [ ] AnomalyReadout flags anomalous states correctly on synthetic data
- [ ] PatternReadout returns valid probability distributions over pattern classes
- [ ] All three readouts share the same reservoir state (not separate reservoirs)
- [ ] Ridge regression training converges on synthetic data
- [ ] All tests pass (pytest tests/learning/test_readouts.py)

## Blocked by

#9 (ESN core must exist first)

## Program Design

_CLASSIFY: new feature (pure-library module on top of existing ESN core; no orchestration/control-flow change → call-stack tree omitted)_

### File-tree diff

```diff
 src/cortex/learning/
+  ├── readouts.py               # NEW — SalienceReadout, AnomalyReadout, PatternReadout
 ~  └── __init__.py              # MODIFIED — export the three readout classes
 tests/learning/
+  └── test_readouts.py          # NEW — per-readout training/prediction + cross-readout independence
```

### Type signatures

```python
# src/cortex/learning/readouts.py

class SalienceReadout(Readout):
    """Binary salience flag over a continuous 0-1 score, ridge-trained."""

    def __init__(self, *, threshold: float = 0.5, ridge_lambda: float = 1e-6) -> None
    def score(self, state: NDArray[np.float64]) -> float:
        """Continuous salience in [0, 1] (float(predict(state)[0]))."""
    def is_salient(self, state: NDArray[np.float64]) -> bool:
        """score(state) >= threshold."""
    # train(states, targets) inherited from Readout — targets in [0, 1]

class AnomalyReadout(Readout):
    """Reconstruction-error anomaly detector: readout trained to reproduce the state."""

    def __init__(self, *, threshold: float = 0.5, ridge_lambda: float = 1e-6) -> None
    def train(self, states: NDArray[np.float64]) -> None:
        """Identity targets: super().train(states, states)."""
    def reconstruction_error(self, state: NDArray[np.float64]) -> float:
        """MSE between state and its reconstruction: mean((state - W_out.T @ state)**2) — exact formula in impl."""
    def is_anomalous(self, state: NDArray[np.float64]) -> bool:
        """reconstruction_error(state) > threshold."""
    # predict(state) inherited from Readout returns the reconstruction (n_reservoir,)

class PatternReadout(Readout):
    """Multi-class probabilities via softmax over ridge logits."""

    def __init__(self, classes: list[str], *, ridge_lambda: float = 1e-6) -> None
    def train(self, states: NDArray[np.float64], labels: NDArray[np.int64]) -> None:
        """One-hot encode labels (n_samples,) in [0, len(classes)) → super().train(states, one_hot)."""
    def get_pattern_probabilities(self, state: NDArray[np.float64]) -> dict[str, float]:
        """Softmax(state @ W_out) → {class_name: probability}, sums to 1.0."""
    # predict(state) inherited from Readout returns raw logits (len(classes),)
```

### Design decisions

- **Subclass `Readout`, don't compose**: all three ARE `Readout`s, so `EchoStateNetwork.register_readout(name, readout)` (contract from #9) accepts them unchanged, and `ESN.read(name)` returns their `predict(state)` output. No facade signature change.
- **`is_salient(state)` not `is_salient(event)`**: the issue body names `event`, but the Boundaries say "Readouts only compute from reservoir state, no external data" — the state vector is the only input. Event→state mapping is the event-bus integration's job (#11).
- **Anomaly = reconstruction error**: `AnomalyReadout.train(states)` fits the readout to reproduce its own input (identity targets). Deviation from the learned normal pattern = MSE between a state and its reconstruction. `threshold` is caller-tuned (data-dependent, no principled default).
- **Pattern labels are integer class indices**: `train(states, labels)` one-hot encodes internally; `classes` list gives the output dict keys and validates label bounds.
- **Softmax applied in `get_pattern_probabilities`, not `predict`**: `predict` stays the raw ridge output (ESN.read contract returns NDArray), the probability mapping is the PatternReadout's own API.
- Thresholds live in constructors (`threshold: float = 0.5`) so `is_salient`/`is_anomalous` are side-effect-free pure queries.
